### ISSUE TITLE
Java exception breaks Mixcr in assemble step

### ISSUE THREAD

**Issue by mvanmoerbeke** [milaboratory/mixcr#501]
## Checklist before submitting the issue:

 -  [X] The issue is strongly related to the MiXCR software
 -  [X]  The issue can be reproduced with [the most recent version](https://github.com/milaboratory/mixcr/releases) of MiXCR
 -  [X] There is no answer to the question in the [official documentation](http://mixcr.readthedocs.io/) and there is no duplicate issue in the [bug tracker](https://github.com/milaboratory/mixcr/issues?utf8=✓&q=)  
 -  [X] Inspection of raw alignments with [exportAlignmentsPretty](http://mixcr.readthedocs.io/en/master/export.html#exporting-well-formatted-alignments-for-manual-inspection) shows that data has the expected architecture, and sample preparation artefacts are not the reason of the problem (if this is the matter of the issue)

## Expected Result

A java exception during the assemble step breaks the analysis before the end. Please find attacted the .txt file with the fastq data from the sample resulting in the error. We didn't observe this error with version 3.0.2 but experience it since version 3.0.5 in samples with low read counts which eventually result in an empty file after the export command.
[01B04.txt](https://github.com/milaboratory/mixcr/files/3012706/01B04.txt)

## Actual Result

``
Preparing for sorting: progress unknown 
Exception in thread "main" picocli.CommandLine$ExecutionException: Error while running command (com.milaboratory.mixcr.cli.CommandAssemble@43d455c9): java.lang.IllegalArgumentException: Writing empty block.
        at picocli.CommandLine.execute(CommandLine.java:1004)
        at picocli.CommandLine.access$900(CommandLine.java:142)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:1199)
        at com.milaboratory.mixcr.cli.Main$1.handle(Main.java:78)
        at com.milaboratory.mixcr.cli.Main$1.handle(Main.java:61)
        at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1075)
        at com.milaboratory.mixcr.cli.Main.handleParseResult(Main.java:83)
        at com.milaboratory.mixcr.cli.Main.main(Main.java:55)
Caused by: java.lang.IllegalArgumentException: Writing empty block.
        at com.milaboratory.mixcr.basictypes.AlignmentsIO.writeBlock(AlignmentsIO.java:139)
        at com.milaboratory.mixcr.basictypes.BasicVDJCAlignmentWriterFactory$Writer.writeSync(BasicVDJCAlignmentWriterFactory.java:221)
        at com.milaboratory.mixcr.basictypes.ClnAWriter.writeAlignmentsAndIndex(ClnAWriter.java:311)
        at com.milaboratory.mixcr.cli.CommandAssemble.run1(CommandAssemble.java:222)
        at com.milaboratory.cli.ACommandWithSmartOverwrite.run0(ACommandWithSmartOverwrite.java:118)
        at com.milaboratory.cli.ACommand.run(ACommand.java:90)
        at picocli.CommandLine.execute(CommandLine.java:996)
        ... 7 more
``

## Exact MiXCR commands 

```
mixcr align --species HomoSapiens --library imgt -OvParameters.geneFeatureToAlign=VRegionWithP -OvParameters.relativeMinScore=0.99 -OdParameters.relativeMinScore=0.99 -OjParameters.relativeMinScore=0.99 ../data/masked/01B04.fastq ../data/mixcr/TRA,TRB/align/01B04.vdjca -f -r ../data/mixcr/TRA,TRB/align/01B04.report.txt

mixcr assemble -OaddReadsCountOnClustering=true -OcloneClusteringParameters.clusteringFilter.specificMutationProbability=1E-2 --write-alignments ../data/mixcr/TRA,TRB/align/01B04.vdjca ../data/mixcr/TRA,TRB/assemble/01B04.clna -f -r ../data/mixcr/TRA,TRB/assemble/01B04.report.txt
```

## MiXCR report files 

The report file from the align step is

```
Analysis date: Wed Mar 27 09:46:12 CET 2019
Input file(s): ../data/masked/01B04.fastq
Output file(s): ../data/mixcr/TRA,TRB/align/01B04.vdjca
Version: 3.0.5; built=Mon Feb 25 00:04:20 CET 2019; rev=af538f7; lib=repseqio.v1.5
Command line arguments: align --species HomoSapiens --library imgt -OvParameters.geneFeatureToAlign=VRegionWithP -OvParameters.relativeMinScore=0.99 -OdParameters.relativeMinScore=0.99 -OjParameters.relativeMinScore=0.99 ../data/masked/01B04.fastq ../data/mixcr/TRA,TRB/align/01B04.vdjca -f -r ../data/mixcr/TRA,TRB/align/01B04.report.txt
Analysis time: 905.00ms
Total sequencing reads: 1
Successfully aligned reads: 1 (100%)
Overlapped: 0 (0%)
Overlapped and aligned: 0 (0%)
Alignment-aided overlaps: 0 (�%)
Overlapped and not aligned: 0 (0%)
TRB chains: 1 (100%)
```

**Comment by PoslavskySV** [comment]
Thanks for reporting! Looks like a bug. Could you share a sample .fastq files, so that we can reproduce the error?

**Comment by mvanmoerbeke** [comment]
Thank you for the quick reply! 

Please find attached some sample .fastq files. 
These are not the most interesting samples due to the low read counts but these result in the mentioned error. 


Van: "Stanislav Poslavsky" <notifications@github.com> 
Aan: "milaboratory/mixcr" <mixcr@noreply.github.com> 
Cc: "Marijke Van Moerbeke" <mvanmoerbeke@openanalytics.eu>, "Author" <author@noreply.github.com> 
Verzonden: Woensdag 27 maart 2019 11:55:36 
Onderwerp: Re: [milaboratory/mixcr] Java exception breaks Mixcr in assemble step (#501) 



Thanks for reporting! Looks like a bug. Could you share a sample .fastq files, so that we can reproduce the error?

**Comment by dbolotin** [comment]
@mvanmoerbeke can you please attach input files through the GitHub interface, I can't download it.

P.S. Sorry for delay, I was on vacation.

**Comment by mvanmoerbeke** [comment]
No problem!
I am not able to attach .fastq files with the GitHub interface as I get the message that the .fastq type is not supported. Can I share the files via eg sendspace of send these to you directly?

**Comment by dbolotin** [comment]
Sure. My email is bolotin dot dmitriy at gmail dot com .

**Comment by dbolotin** [comment]
Pre-release version with the fix:
https://s3.amazonaws.com/files.milaboratory.com/mixcr/mixcr-306b.zip

**Comment by dbolotin** [comment]
was already fixed in f9a35ffce57af82c580a0200247b69dc55c834ab
